causalml/inference/iv/drivlearner.py

```diff
@@ -0,0 +1,864 @@
+            control_name (str or int, optional): name of control group
+        """
+        assert (learner is not None) or (
+            (control_outcome_learner is not None)
+            and (treatment_outcome_learner is not None)
+            and (treatment_effect_learner is not None)
+        )
+
+        if control_outcome_learner is None:
+            self.model_mu_c = deepcopy(learner)
+        else:
+            self.model_mu_c = control_outcome_learner
+
+        if treatment_outcome_learner is None:
+            self.model_mu_t = deepcopy(learner)
+        else:
+            self.model_mu_t = treatment_outcome_learner
+
+        if treatment_effect_learner is None:
+            self.model_tau = deepcopy(learner)
+        else:
+            self.model_tau = treatment_effect_learner
+
+        self.ate_alpha = ate_alpha
+        self.control_name = control_name
+
+        self.propensity_1 = None
+        self.propensity_0 = None
+        self.propensity_assign = None
+
+    def __repr__(self):
+        return (
+            "{}(control_outcome_learner={},\n"
+            "\ttreatment_outcome_learner={},\n"
+            "\ttreatment_effect_learner={})".format(
+                self.__class__.__name__,
+                self.model_mu_c.__repr__(),
+                self.model_mu_t.__repr__(),
+                self.model_tau.__repr__(),
+            )
+        )
+
+    def fit(
+        self, X, assignment, treatment, y, p=None, pZ=None, seed=None, calibrate=True
+    ):
+        """Fit the inference model.
+
+        Args:
+            X (np.matrix or np.array or pd.Dataframe): a feature matrix
+            assignment (np.array or pd.Series): a (0,1)-valued assignment vector
+            treatment (np.array or pd.Series): a treatment vector
+            y (np.array or pd.Series): an outcome vector
+            p (2-tuple of np.ndarray or pd.Series or dict, optional): The first (second) element corresponds to
+                unassigned (assigned) units. Each is an array of propensity scores of float (0,1) in the single-treatment
+                case; or, a dictionary of treatment groups that map to propensity vectors of float (0,1). If None will run
+                ElasticNetPropensityModel() to generate the propensity scores.
+            pZ (np.array or pd.Series, optional): an array of assignment probability of float (0,1); if None
+                will run ElasticNetPropensityModel() to generate the assignment probability score.
+            seed (int): random seed for cross-fitting
+        """
+        X, treatment, assignment, y = convert_pd_to_np(X, treatment, assignment, y)
+        check_treatment_vector(treatment, self.control_name)
+        self.t_groups = np.unique(treatment[treatment != self.control_name])
+        self.t_groups.sort()
+        self._classes = {group: i for i, group in enumerate(self.t_groups)}
+
+        cv = KFold(n_splits=3, shuffle=True, random_state=seed)
```

Comment:
It'd be nice to explain how we're using these 3-folds: one for the propensity score, one for the outcome, and last one for the treatment effect.

Reply:
Thanks for the suggestion! I added a comment in the code to explain this.